extensions/hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/HadoopFileSourceFactory.java

```diff
@@ -197,7 +199,13 @@ public String format() {
 
             Class<?> clazz = csvFileFormat.clazz();
             if (clazz != null) {
-                job.getConfiguration().set(CSV_INPUT_FORMAT_BEAN_CLASS, clazz.getCanonicalName());
+                job.getConfiguration().set(CSV_INPUT_FORMAT_BEAN_CLASS, clazz.getName());
+            }
+            List<String> fieldList = csvFileFormat.stringArrayFieldList();
+            if (fieldList != null) {
+                for (int i = 0; i < fieldList.size(); i++) {
+                    job.getConfiguration().set(CSV_INPUT_FORMAT_FIELD_LIST_PREFIX + i, fieldList.get(i));
```

Comment:
Maybe use `org.apache.hadoop.conf.Configuration#setStrings(String, String[])`

Reply:
Looked good, but it trivially joins the strings using `","` and the uses `String.split(",")`. Though not common, a field name can contain a `,`, especially in CSV when headers are sometimes typed in Excel...